src/signup.js

```diff
@@ -0,0 +1,110 @@
+const emailContainer=document.querySelector('#email-input-container')
+const emailInput = emailContainer.querySelector('#email')
+const emailMsg = emailContainer.querySelector('#email-msg')
+const nameContainer=document.querySelector('#name-input-container')
+const nameInput = nameContainer.querySelector('#name')
+const nameMsg = nameContainer.querySelector('#name-msg')
+const passwordContainer=document.querySelector('#password-input-container')
+const passwordInput = passwordContainer.querySelector('#password')
+const passwordMsg = passwordContainer.querySelector('#password-msg')
+const passwordCheckContainer=document.querySelector('#password-check-input-container')
+const passwordCheckInput = passwordCheckContainer.querySelector('#password-check')
+const passwordCheckMsg = passwordCheckContainer.querySelector('#password-check-msg')
+const button = document.querySelector('#signup-button')
+
+
+emailInput.addEventListener('focusout', emailValidCheck);
+nameInput.addEventListener('focusout', nameValidCheck);
+passwordInput.addEventListener('focusout', passwordValidCheck);
+passwordCheckInput.addEventListener('focusout', passwordCheckValidCheck);
+
+emailInput.addEventListener('focusout', buttonActivate);
+nameInput.addEventListener('focusout', buttonActivate);
+passwordInput.addEventListener('focusout', buttonActivate);
+passwordCheckInput.addEventListener('focusout', buttonActivate);
+
+let emailValid=false;
+let nameValid=false;
+let passwordValid=false;
+let passwordCheckValid=false;
+
+function buttonActivate() {
+
+    if(emailValid&&nameValid&&passwordValid&&passwordCheckValid) { 
+      button.disabled = false;
+    } 
+    else { 
+      button.disabled = true; 
+    }
+}
```

Comment:
P3:
아래 코드도 동일동작 합니다.

```suggestion
function buttonActivate() {
    const isValid = emailValid&&nameValid&&passwordValid&&passwordCheckValid;
    button.disabled = !isValid;
}
```